src/merkle/path_set.rs

```diff
@@ -78,15 +81,18 @@ impl MerklePathSet {
     /// Returns an error if:
     /// * The specified index not valid for the depth of structure.
     /// * Node of the requested path does not exist in the set.
-    pub fn get_path(&self, depth: u32, index: u64) -> Result<MerklePath, MerkleError> {
-        if index >= 2u64.pow(self.total_depth) {
-            return Err(MerkleError::InvalidIndex(self.total_depth, index));
+    pub fn get_path(&self, node: NodeIndex) -> Result<MerklePath, MerkleError> {
```

Comment:
nit: should the parameter be named `index` rather than `node`?

Reply:
Done!